include/pcp/filter/statistical_outlier_filter.hpp

```diff
@@ -0,0 +1,141 @@
+#ifndef PCP_FILTER_STATISTICAL_OUTLIER_FILTER_HPP
+#define PCP_FILTER_STATISTICAL_OUTLIER_FILTER_HPP
+/**
+ * @file
+ * @ingroup filter
+ */
+
+#include <algorithm>
+#include <iterator>
+#include <map>
+#include <pcp/kdtree/linked_kdtree.hpp>
+#include <vector>
+namespace pcp {
+
+namespace statistical_outlier_filter {
+/**
+ * @brief parameters for the filter
+ * @tparam ScalarType Scalar type for the parameter variables (float,double,etc)
+ */
+template <class ScalarType>
+struct construction_params_t
+{
+    using scalar_type                         = ScalarType;
+    scalar_type std_dev_multiplier_threshold_ = static_cast<float>(1);
+    std::size_t mean_k_                       = 1;
+};
+} // namespace statistical_outlier_filter
+
+/**
+ * @brief The algorithm iterates through the entire input twice:
+ * During the first iteration it will compute the average distance that each point has to its
+ * nearest k neighbors. The value of k can be set using mean_k_ Next, the mean and standard
+ * deviation of all these distances are computed in order to determine a distance threshold. The
+ * distance threshold will be equal to: mean + stddev_mult * stddev. ' The multiplier for the
+ * standard deviation can be set using the std_dev_multiplier_treshold in the params.
+ *
+ * @tparam Element Element type
+ * @tparam ParametersType type of the parameters (double,float)
+ * @tparam PointMap Type satisfying PointMap concept
+ * @tparam KnnMap Type satisfying KnnMap concept
+ */
+template <class Element, class ParametersType, class PointMap, class KnnMap>
+
+class basic_statistical_outlier_filter_t
```

Comment:
Should not have the "basic_" prefix. The reason is discussed in a previous comment.